source/adapters/opencl/device.cpp

```diff
@@ -563,6 +563,24 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
                         UR_MEMORY_ORDER_CAPABILITY_FLAG_RELEASE |
                         UR_MEMORY_ORDER_CAPABILITY_FLAG_ACQ_REL |
                         UR_MEMORY_ORDER_CAPABILITY_FLAG_SEQ_CST;
+    } else {
+      // FIXME: Special case for Intel FPGA driver which is currently an OpenCL
+      // 1.2 device but is more capable than the default. This is a temporary
+      // work around until the Intel FPGA driver is updated to OpenCL 3.0.
+      size_t deviceNameSize;
+      CL_RETURN_ON_FAILURE(
+          clGetDeviceInfo(cl_adapter::cast<cl_device_id>(hDevice),
+                          CL_DEVICE_NAME, 0, nullptr, &deviceNameSize));
+      std::string deviceName(deviceNameSize, '\0');
+      CL_RETURN_ON_FAILURE(clGetDeviceInfo(
+          cl_adapter::cast<cl_device_id>(hDevice), CL_DEVICE_NAME,
+          deviceNameSize, deviceName.data(), nullptr));
+      if (deviceName.find("Intel(R) FPGA SDK for OpenCL(TM)") !=
+              std::string::npos ||
+          deviceName.find("Intel(R) FPGA Emulation Platform for OpenCL(TM)") !=
+              std::string::npos) {
+        URCapabilities = UR_MEMORY_SCOPE_CAPABILITY_FLAG_DEVICE;
```

Comment:
should maybe be
```suggestion
        URCapabilities |= UR_MEMORY_SCOPE_CAPABILITY_FLAG_DEVICE;
```

Reply:
Thanks, I refactored this a bit because I implemented the wrong workaround. I've made sure to `|=` the value. Please check again.